swagger-api/swagger-ui
# fix: add client_id and client_secret to form when type is request-body

In an OAuth2 password flow, when the type is `request-body`, the `client_id` and `client_secret` were being passed in an `Authorization` header and not in the request body.

### Description

Refactored the `authorizePassword` function to properly handle the `request-body` case.

### Motivation and Context




Fixes #4192 

### How Has This Been Tested?




1. Load the following sample API definition into Swagger UI: http://flask-restplus-example-server.herokuapp.com/api/v1/swagger.json
2. Click the Authorize button
3. Enter a username/password/client ID/client secret
4. Ensure that the type is set to Request Body
5. Click the Authorize button  (note: the actual authorization will fail, but for the purposes of this PR it is not an issue)
6. Open the DevTools Network panel and check the form data in the request includes the username/password/client ID/client secret (see screenshot below)

### Screenshots (if appropriate):
![swagger_ui_and_actions_js_ _swagger-ui](https://user-images.githubusercontent.com/289327/36080392-8b1b5b9e-0fe3-11e8-8527-bd2958ffd866.png)

## Checklist



### My PR contains... 


### My changes...

### Documentation

### Automated tests
